.github/workflows/build.yml

```diff
@@ -53,21 +49,42 @@ jobs:
         id: login-ecr
         uses: aws-actions/amazon-ecr-login@v1
         with:
-          mask-password: 'true'
+          mask-password: "true"
+
+      - name: Set image tag in output
+        id: "set-image-tag"
+        run: |-
+          echo "image_tag=$(
+          if ([ "${{ github.ref_name }}" == "main" ] || [ "${{ github.ref_name }}" == "master" ]) && [ "${{ github.event_name }}" == "push" ]; then
+            echo DEV-"${{ github.sha }}"
+          elif [ "${{ github.event_name }}" == "release" ] && [ "${{ github.event.action }}" == "released" ]; then
+            echo PROD-"${{ github.ref_name }}"
+          elif [ "${{ github.event_name }}" == "pull_request" ] && ([ "${{ github.event.action }}" == "opened" ] || [ "${{ github.event.action }}" == "synchronize" ] || [ "${{ github.event.action }}" == "closed" ]); then
+            echo PREVIEW-"${{ github.head_ref }}"
+          else
+            echo unknown
+          fi
+          )" >> $GITHUB_OUTPUT
+
+      - name: Echo image tag
+        env:
+          TAG: "${{ steps.set-image-tag.outputs.image_tag }}"
+        run: |-
+          echo "$TAG"
 
       - name: Build, tag, and push image to Amazon ECR
-        if: github.event.action != 'closed'
+        if: github.event.action != 'closed' &&  steps.set-image-tag.outputs.image_tag  != 'unknown'
```

Comment:
Lurer litt på om vi burde sørge for at workflowen feiler eksplisitt hvis vi havner med `image_tag=unknown`, for nå tror jeg den i så fall bare ikke vil lage og pushe Docker-image, og så vil det kanskje feile hardt i deploy-workflowen? Tror det kan gjøres på mange måter, for eksempel med et step til slutt som sier noe ala:
```yml
- name: Terraform Plan Status for merge
  if: steps.set-image-tag.outputs.image_tag == 'unknown'
  run: exit 1
```

Reply:
Lurt!